File: app/src/main/java/com/sagalogistics/backend/database/RepositoryFactoryFirebase.java
package com.sagalogistics.backend.database;

import androidx.annotation.NonNull;

public class RepositoryFactoryFirebase implements RepositoryFactory{
    @NonNull
    @Override
    public ItemDAO createItemDAO() {
        return new ItemDAOFirebase();
    }

    @NonNull
    @Override
    public OrderDAO createOrderDAO() {
        return new OrderDAOFirebase();
    }
}

File: app/src/main/java/com/sagalogistics/backend/database/RepositoryFactoryFirebase.kt
package com.sagalogistics.backend.database

class RepositoryFactoryFirebase : RepositoryFactory {
    override fun createItemDAO(): ItemDAO {
        return ItemDAOFirebase()
    }

    override fun createOrderDAO(): OrderDAO {
        return OrderDAOFirebase()
    }

    override fun createBarDAO(): BarDAO {
        return BarDAOFirebase()
    }

    override fun createUserDAO(): UserDAO {
        return UserDAOFirebase()
    }
}